addons/mrp/models/mrp_production.py

```diff
@@ -741,6 +741,13 @@ def write(self, vals):
 
     @api.model
     def create(self, values):
+        # Remove from `move_finished_ids` the by-product moves and then move `move_byproduct_ids`
+        # into `move_finished_ids` to avoid duplicate and inconsistency.
+        if values.get('move_finished_ids', False):
```

Comment:
I think i will be cleaner to do the same process than the move_without_package magic 
https://github.com/odoo/odoo/blob/5e5de2b2b64fd8cb259e72fc1a97dc754da871ad/addons/stock/models/stock_picking.py#L778-L805

Re-computing the byproduct_move_ids each time it's updated